### PR TITLE
Feature refector dataProviders

### PR DESCRIPTION
**Pull request by BrianCodeman**
1.Remove IRepository from All DataProviders
2.Move All Add/Delete method to DataProviders
2.Add Update Interface to DataProviders

### CODE REVIEW

**Comment by BrianCodeman** [comment]
Due to "Update" implementation has not been reviewed and merged, the CI failed here. It's normal and should not be an issue. It will pass when IRepository Update brand been merged.

**Comment by BrianCodeman** [comment]
Now the IRepository Update branch has merged and CI process has passed.

File: src/SugarChat.Message/Commands/Users/UpdateUserCommand.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+
+namespace SugarChat.Message.Commands.Users
+{
+    public class UpdateUserCommand
```
Comment (mayuanyang) [inline src/SugarChat.Message/Commands/Users/UpdateUserCommand.cs:5]: 需要implement ICommand

File: src/SugarChat.Message/Events/Users/UserUpdatedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SugarChat.Message.Events.Users
+{
+    public class UserUpdatedEvent : EventBase
```
Comment (mayuanyang) [inline src/SugarChat.Message/Events/Users/UserUpdatedEvent.cs:3]: implement IEvent

**Review by mayuanyang — CHANGES_REQUESTED** [review]
留了一点comment，另外需要加点tests？

File: src/SugarChat.Message/Events/Users/UserUpdatedEvent.cs (hunk 2 above)
Comment (BrianCodeman) [reply src/SugarChat.Message/Events/Users/UserUpdatedEvent.cs:3]: EventBase implements IEvent

File: src/SugarChat.Message/Commands/Users/UpdateUserCommand.cs (hunk 1 above)
Comment (BrianCodeman) [reply src/SugarChat.Message/Commands/Users/UpdateUserCommand.cs:5]: 这个漏了，感谢review

**Comment by BrianCodeman** [comment]
Tests will be added soon in next pr, and this one will mostly focus on defining resonable Service and Provider Interfaces, and part of implementation of course.

**Review by ppXD — APPROVED** [review]
Approved

**Pull request merged**